service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -94,11 +96,10 @@ public MetadbSample fetchAndLoadSampleDetails(MetadbSample sample) throws Except
     @Override
     public MetadbSample getMetadbSample(UUID metadbSampleId) throws Exception {
         MetadbSample sample = sampleRepository.findSampleById(metadbSampleId);
-        sample.setSampleMetadataList(sampleRepository.findSampleMetadataListBySampleId(metadbSampleId));
-        for (SampleMetadata s: sample.getSampleMetadataList()) {
-            s.setMetaDbPatientId(patientService.getPatientIdBySample(metadbSampleId));
-            s.setMetaDbSampleId(metadbSampleId);
+        if (sample == null) {
+            return null;
         }
+        sample.setSampleMetadataList(sampleRepository.findSampleMetadataListBySampleId(metadbSampleId));
```

Comment:
need to fetch the sample's patient here still

Reply:
[reference](https://github.com/mskcc/cmo-metadb/pull/468/files#diff-14fbd1dc6e948619c6d34efb8f41005965101954a8895440857979d916eb6b57L130)